tests/nose_runner.py

```diff
@@ -1,134 +1,16 @@
-import multiprocessing
 import os
-import sys
 
-import nose
-import numpy
-import yaml
+import pytest
 
 from yt.config import ytcfg
-from yt.utilities.answer_testing.framework import AnswerTesting
 
-numpy.set_printoptions(threshold=5, edgeitems=1, precision=4)
-
-
-class NoseWorker(multiprocessing.Process):
-    def __init__(self, task_queue, result_queue):
-        multiprocessing.Process.__init__(self)
-        self.task_queue = task_queue
-        self.result_queue = result_queue
-
-    def run(self):
-        proc_name = self.name
-        while True:
-            next_task = self.task_queue.get()
-            if next_task is None:
-                print("%s: Exiting" % proc_name)
-                self.task_queue.task_done()
-                break
-            print("%s: %s" % (proc_name, next_task))
-            result = next_task()
-            self.task_queue.task_done()
-            self.result_queue.put(result)
-            if next_task.exclusive:
-                print("%s: Exiting (exclusive)" % proc_name)
-                break
-        return
-
-
-class NoseTask(object):
-    def __init__(self, job):
-        argv, exclusive = job
-        self.argv = argv
-        self.name = argv[0]
-        self.exclusive = exclusive
-
-    def __call__(self):
-        test_dir = ytcfg.get("yt", "test_data_dir")
-        answers_dir = os.path.join(test_dir, "answers")
-        if "--with-answer-testing" in self.argv and not os.path.isdir(
-            os.path.join(answers_dir, self.name)
-        ):
-            nose.run(
-                argv=self.argv + ["--answer-store"],
-                addplugins=[AnswerTesting()],
-                exit=False,
-            )
-        if os.path.isfile("{}.xml".format(self.name)):
-            os.remove("{}.xml".format(self.name))
-        nose.run(argv=self.argv, addplugins=[AnswerTesting()], exit=False)
-        return ""
-
-    def __str__(self):
-        return "WILL DO self.name = %s" % self.name
-
-
-def generate_tasks_input():
-    pyver = "py{}{}".format(sys.version_info.major, sys.version_info.minor)
-    test_dir = ytcfg.get("yt", "test_data_dir")
-    answers_dir = os.path.join(test_dir, "answers")
-    tests = yaml.load(open("tests/tests.yaml", "r"), Loader=yaml.FullLoader)
-
-    base_argv = ["-s", "--nologcapture", "--with-xunit"]
-
-    base_answer_argv = [
-        "--local-dir=%s" % answers_dir,
-        "--with-answer-testing",
+if __name__ == "__main__":
+    pytest_args = [
+        f"--local-dir={os.path.join(ytcfg.get('yt', 'test_data_dir'), 'answers')}",
+        "-c=pytest_answer.ini",
+        "--junitxml=unittests.xml",
         "--answer-big-data",
-        "--local",
+        f"-n {int(os.environ.get('NUM_WORKERS', 6))}",
+        "--dist loadscope",
```

Comment:
```suggestion
        "--dist=loadscope",
```
Otherwise `pytest.main()` treats it as a separate arg and tries to execute test called `loadscope` which doesn't exist and thus no tests are being run on Jenkins...

Reply:
Gotcha, thanks! I'll also regenerate the answers, just to be safe